### DISCUSSION TITLE
Its Show always first Object - My json file size is 634 MB

### DISCUSSION THREAD

**Discussion by saaqeb** [halaxa/json-machine#85] [General]
Its Show always first Object - My json file size is 634 MB

`
require_once './vendor/autoload.php';
$testfile = './propertycatalog.expediacollect.en-US.jsonl';

use JsonMachine\JsonDecoder\ExtJsonDecoder;
use JsonMachine\Items;

$objects = Items::fromFile($testfile, ['decoder' => new ExtJsonDecoder(true)]);

foreach ($objects as $id => $user) {
    // just process $user as usual
		
	print"<pre>";
	print_r($user);
	print"</pre>";
	print"<br/>------------------------<br/><br/>";
   
}

`

**Comment by halaxa** [comment]
From the jsonl extension, I guess you use json lines format (https://jsonlines.org/). It itself is not valid json document, but every line is a standalone document itself. That's why JSON Machine parses only the first line - the first valid document. JSONL is usually parsed by reading a document one line at a time and simply `json_decode` it.
